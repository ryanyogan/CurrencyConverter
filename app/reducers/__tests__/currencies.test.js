import reducer from '../currencies';
import {
  getInitialConversion,
  changeCurrencyAmount,
  swapCurrency,
  changeBaseCurrency,
  changeQuoteCurrency,
} from '../../actions/currencies';

it('sets initial state', () => {
  const expected = {
    baseCurrency: 'USD',
    quoteCurrency: 'CAD',
    amount: 100,
    conversions: {},
    error: null,
  };
  const result = reducer(undefined, {});
  expect(result).toEqual(expected);
});

it('snapshots the initialState', () => {
  expect(reducer(undefined, {})).toMatchSnapshot();
});

it('sets nested data on initialState', () => {
  const expected = {
    baseCurrency: 'USD',
    quoteCurrency: 'CAD',
    amount: 100,
    conversions: {
      USD: {
        isFetching: true,
        date: '',
        rates: {},
      },
    },
    error: null,
  };
  const response = reducer(undefined, getInitialConversion());
  expect(response).toEqual(expected);
});

it('snapshots nested data on initialState', () => {
  expect(reducer(undefined, getInitialConversion())).toMatchSnapshot();
});

it('defaults change currency amount to 0', () => {
  expect(reducer(undefined, changeCurrencyAmount())).toMatchSnapshot();
});

it('changes the currency amount', () => {
  expect(reducer(undefined, changeCurrencyAmount(1000))).toMatchSnapshot();
});

it('swaps base and quote currencies', () => {
  expect(reducer(undefined, swapCurrency())).toMatchSnapshot();
});

it('changes the base currency', () => {
  expect(reducer(undefined, changeBaseCurrency('CZK'))).toMatchSnapshot();
});

it('changes the quote currency', () => {
  expect(reducer(undefined, changeQuoteCurrency('GBP'))).toMatchSnapshot();
});
