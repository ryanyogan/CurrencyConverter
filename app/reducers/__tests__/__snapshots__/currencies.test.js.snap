// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changes the base currency 1`] = `
Object {
  "amount": 100,
  "baseCurrency": "CZK",
  "conversions": Object {
    "CZK": Object {
      "date": "",
      "isFetching": true,
      "rates": Object {},
    },
  },
  "error": null,
  "quoteCurrency": "CAD",
}
`;

exports[`changes the currency amount 1`] = `
Object {
  "amount": 1000,
  "baseCurrency": "USD",
  "conversions": Object {},
  "error": null,
  "quoteCurrency": "CAD",
}
`;

exports[`changes the quote currency 1`] = `
Object {
  "amount": 100,
  "baseCurrency": "USD",
  "conversions": Object {
    "GBP": Object {
      "date": "",
      "isFetching": true,
      "rates": Object {},
    },
  },
  "error": null,
  "quoteCurrency": "GBP",
}
`;

exports[`defaults change currency amount to 0 1`] = `
Object {
  "amount": 0,
  "baseCurrency": "USD",
  "conversions": Object {},
  "error": null,
  "quoteCurrency": "CAD",
}
`;

exports[`snapshots nested data on initialState 1`] = `
Object {
  "amount": 100,
  "baseCurrency": "USD",
  "conversions": Object {
    "USD": Object {
      "date": "",
      "isFetching": true,
      "rates": Object {},
    },
  },
  "error": null,
  "quoteCurrency": "CAD",
}
`;

exports[`snapshots the initialState 1`] = `
Object {
  "amount": 100,
  "baseCurrency": "USD",
  "conversions": Object {},
  "error": null,
  "quoteCurrency": "CAD",
}
`;

exports[`swaps base and quote currencies 1`] = `
Object {
  "amount": 100,
  "baseCurrency": "CAD",
  "conversions": Object {},
  "error": null,
  "quoteCurrency": "USD",
}
`;
